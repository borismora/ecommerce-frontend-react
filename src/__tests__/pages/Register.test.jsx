import { render, screen } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import Register from '../../pages/Register';

vi.mock('../../components/Auth/RegisterForm', () => ({
  default: () => <div data-testid="register-form">Mock RegisterForm</div>,
}));

describe('Register Page', () => {
  it('renders RegisterForm component', () => {
    render(<Register />);
    expect(screen.getByTestId('register-form')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Register />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with different props', () => {
    const { rerender } = render(<Register />);
    expect(screen.getByTestId('register-form')).toBeInTheDocument();

    rerender(<Register />);
    expect(screen.getByTestId('register-form')).toBeInTheDocument();
  });
});
