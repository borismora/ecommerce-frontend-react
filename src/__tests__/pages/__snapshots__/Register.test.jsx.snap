// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Register Page > matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="register-form"
  >
    Mock RegisterForm
  </div>
</DocumentFragment>
`;
